src/main/java/com/teamcautionrobotics/robot/DriveBase.java

```diff
@@ -61,6 +75,46 @@ public void drive(double speed) {
         drive(speed, speed);
     }
 
+    public double getJerkLimit() {
+        return jerkLimit;
+    }
+
+    public void setJerkLimit(double jerkLimit) {
+        this.jerkLimit = jerkLimit;
+    }
+
+    public void driveSmoothly(double leftInput, double rightInput) {
+        updateDerivatives(leftInput, rightInput);
+        double leftPower = leftInput;
+        double rightPower = rightInput;
+
+        // limit jerk for each side if predicted jerk is too high
+        if (leftInputDerivative > jerkLimit) {
```

Comment:
You will need some absolute values here, otherwise this only limits jerk in one direction.